src/result.hpp

```
@@ -0,0 +1,37 @@
+    EbpfDomain pre;
+    std::optional<VerificationError> error;
+    EbpfDomain post;
+};
+
+struct AnalysisResult {
+    std::map<Label, InvariantMapPair> invariants;
+    bool failed = false;
+    int max_loop_count{};
+    Interval exit_value = Interval::top();
+
+    bool is_valid_after(const Label& label, const StringInvariant& state) const;
```

Clang (14.0.6)

[warning] function 'is_valid_after' should be marked [[nodiscard]]

(modernize-use-nodiscard)